lib/handler-storage.js

```diff
@@ -47,8 +47,8 @@ class HandlerStorage {
     }
 
     const isMergedTree = constraintsNames.includes(httpMethodStrategy.name)
-    if (!isMergedTree && this.handlers.length >= 32) {
-      throw new Error('find-my-way supports a maximum of 32 route handlers per node when there are constraints, limit reached')
+    if (!isMergedTree && this.handlers.length >= 31) {
+      throw new Error('find-my-way supports a maximum of 31 route handlers per node when there are constraints, limit reached')
```

Comment:
```suggestion
    if (!isMergedTree && this.handlers.length >= 33) {
      throw new Error('find-my-way supports a maximum of 32 route handlers per node when there are constraints, limit reached')
```

Reply:
Isn't this the opposite of what we want? 

The logic on main throws when we add 33 handlers, but routing breaks with 32.

So what I was trying to achieve with changing to 31 is to throw when we know routing breaks. The reason we need to use 31 here, is because the check is done before we have added current handler to the list.